driver/generic/ili9xxx.py

```diff
@@ -0,0 +1,148 @@
+_MADCTL_BGR = const(0x08)  # colors are BGR (not RGB)
+_MADCTL_RTL = const(0x04)  # refresh right to left
+
+_MADCTL_ROTS = (
+    const(_MADCTL_MX),  # 0 = portrait
+    const(_MADCTL_MV),  # 1 = landscape
+    const(_MADCTL_MY),  # 2 = inverted portrait
+    const(_MADCTL_MX | _MADCTL_MY | _MADCTL_MV),  # 3 = inverted landscape
+)
+
+
+class Ili9341_hw(St77xx_hw):
+    def __init__(self, res, **kw):
+        super().__init__(
+            res=res,
+            suppRes=[
+                (240, 320),
+            ],
+            model=None,
+            suppModel=None,
+            bgr=True,
+            **kw,
+        )
+
+    def config_hw(self):
+        self._run_seq(
+            [
+                (_SLPOUT, None, 100),
+                (_PWCTRB, b"\x00\xC1\x30"),  # Pwr ctrl B
+                (_POSC, b"\x64\x03\x12\x81"),  # Pwr on seq. ctrl
+                (_DTCA, b"\x85\x00\x78"),  # Driver timing ctrl A
+                (_PWCTRA, b"\x39\x2C\x00\x34\x02"),  # Pwr ctrl A
+                (_PUMPRC, b"\x20"),  # Pump ratio control
+                (_DTCB, b"\x00\x00"),  # Driver timing ctrl B
+                (_PWCTR1, b"\x23"),  # Pwr ctrl 1
+                (_PWCTR2, b"\x10"),  # Pwr ctrl 2
+                (_VMCTR1, b"\x3E\x28"),  # VCOM ctrl 1
+                (_VMCTR2, b"\x86"),  # VCOM ctrl 2
+                # (_MADCTL, self.rotation"),  # Memory access ctrl  # TODO: rotation?
+                (_VSCRSADD, b"\x00"),  # Vertical scrolling start address
+                (_PIXFMT, b"\x55"),  # COLMOD: Pixel format
+                (_FRMCTR1, b"\x00\x18"),  # Frame rate ctrl
+                (_DFUNCTR, b"\x08\x82\x27"),
+                (_ENABLE3G, b"\x00"),  # Enable 3 gamma ctrl
+                (_GAMMASET, b"\x01"),  # Gamma curve selected
+                (
+                    _GMCTRP1,
+                    b"\x0F\x31\x2B\x0C\x0E\x08\x4E\xF1\x37\x07\x10\x03\x0E\x09\x00",
+                ),
+                (
+                    _GMCTRN1,
+                    b"\x00\x0E\x14\x03\x11\x07\x31\xC1\x48\x08\x0F\x0C\x31\x36\x0F",
+                ),
+                (_SLPOUT, None, 100),
+                (_DISPLAY_ON, None),
+            ]
+        )
+
+    def apply_rotation(self, rot):
+        self.rot = rot
+        if (self.rot % 2) == 0:
+            self.width, self.height = self.res
+        else:
+            self.height, self.width = self.res
+        self.write_register(
+            _MADCTL,
+            bytes([(_MADCTL_BGR if self.bgr else 0) | _MADCTL_ROTS[self.rot % 4]]),
+        )
+
+
+class Ili9341(Ili9341_hw, St77xx_lvgl):
+    def __init__(self, res, doublebuffer=True, factor=4, **kw):
+        """See :obj:`Ili9341_hw` for the meaning of the parameters."""
+        import lvgl as lv
+
+        Ili9341_hw.__init__(self, res=res, **kw)
+        St77xx_lvgl.__init__(self, doublebuffer, factor)
+        self.disp_drv.color_format = lv.COLOR_FORMAT.NATIVE_REVERSE
+        self.disp_drv.register()
```

Comment:
Without running `register()` again, the change in color_format does not take effect. I think the most appropriate thing would be to call the python equivalent of `lv_disp_drv_update`, but I don't think it's bound? I'm not sure on how to solve this.

Reply:
> I think the most appropriate thing would be to call the python equivalent of `lv_disp_drv_update`, but I don't think it's bound?

Since the binding is auto generated from LVGL's C headers, virtually all LVGL API is bound.  
`lv_disp_drv_update` receives `lv_disp_t` as an argument, therefore it is considered a member of `lv.disp_t`.

Something like this might work:
```python
disp = lv.disp_get_default()
disp.drv_update(new_drv)
```

Reply:
I tried that, but it didn't work. Any other ideas?

Reply:
I also tried:

```
self.disp = self.disp_drv.register()
...
self.disp_drv.color_format = lv.COLOR_FORMAT.NATIVE_REVERSE
self.disp.drv_update(self.disp_drv)
```
but that also didn't work; is `disp_drv_update` broken?

Reply:
setting `bgr=False` and reversing my logic in `apply_rotation` makes everything work properly without modifying other classes, so lets just go with that for now.